trie/proof_generator_test.go

```diff
@@ -182,7 +182,8 @@ func TestSerialiseBlockWitness(t *testing.T) {
 	if err := bwb.WriteTo(&b); err != nil {
 		t.Errorf("Could not make block witness: %v", err)
 	}
-	expected := common.FromHex("0xa76862616c616e6365730065636f64657300666861736865731822646b65797300666e6f6e63657300697374727563747572650b6676616c75657300582023181a62d35fe01562158be610f84e047f99f5e74d896da21682d925964ece3a0601024704010402040304")
+
+	expected := common.FromHex("0xa76862616c616e6365730065636f64657300666861736865731822646b65797300666e6f6e63657300697374727563747572650b6676616c756573005820858f70a4b1e6aa71a7edc574d2ca946495a038aa37ce13dc7b7ed15661a6ff2f0601024704010402040304")
```

Comment:
Why did this value change?

Reply:
we no longer treat `[]byte("val1")` and `[]byte("val2")` in `tr.Update(...)` calls (lines 174-175) as RLP-encoded values, so we add a double prefix to them when hashing, so they produce different hash values.